### ISSUE TITLE
Move Inline edit inline table to "experimental"

### ISSUE THREAD

**Issue by seanforyou23** [patternfly/patternfly-react#2620]
Currently, there is a disconnect with [Inline edit inline table](https://patternfly-react.surge.sh/patternfly-4/inline%20table/inlineedit/). It's being actively used in design artifacts for PF4 apps, but there's no direct support from the PF team and this component has problems.

The component doesn't work when pulled into an app built using TypeScript - type errors. Consuming it as a JavaScript module will silence the type errors, but in turn, lead our product down a path that isn't viewed as healthy. Beyond type errors, the component demo code contains other generic problems which further indicates that this code isn't being tested or maintained.

TIL "extensions" are not patternfly ideas in general and are almost always _only_ supported by the community. Further, these types of components are described in the following manner

> Extension components do not undergo the same rigorous design or coding review process as core PatternFly components.

This guides developers in between a rock and hard place with multiple unattractive paths forward.

Outstanding questions:

- Can we move inline edit table to "experimental" so that there is at least a possibility that the type errors can be fixed in a timeframe that is relevant for our product?
- Can we remove extensions from the react workspace as not to mislead designers into suggesting patterns of which we have no workable implementation?
- Can we consider adding this component functionality as part of the @patternfly/react-table pkg?

A relevant comment recent work item: https://github.com/patternfly/patternfly-react/pull/2360#issuecomment-516430434

We are sorta blocked on this at the moment. Don't necessarily need the entire problem fixed immediately, mostly just want to understand how we can best move forward in the short term and do so in a way that isn't out of alignment with the direction of the PF framework. Please let me know if I can clarify anything - thank you! 🙏

**Comment by janwright73** [comment]
@LHinson fuse online wants this for 7.5 release -  @seanforyou23 knows more.

**Comment by dgutride** [comment]
@seanforyou23  @priley86  - can somebody find the initial PR/issue that landed this in an extension to start?  I'm trying to remember the history, but in general - if this is needed by multiple products and belongs in PatternFly as a pattern, I think we should consider this request.

**Comment by rareddy** [comment]
For Data Virtualization in Syndesis, we want to start the development with "View Output" screen, which is using this component per UX team's mockups, any prioritization on this component would be greatly appreciated.

**Comment by priley86** [comment]
opened a PR temporarily to improve Gatsby docs on extensions...
https://github.com/patternfly/patternfly-react/pull/2637

I'm happy to see `@patternfly/react-inline-edit-extension` as an extension or as an experimental component until it is officially added in Core.

`@patternfly/react-virtualized-extension` is being used actively downstream, but it is technically still can extension. It is also up for design/a11y review in Core in the future.

**Comment by rachael-phillips** [comment]
I was talking with @rareddy who needs the above inline-table editor mentioned in the GitHub issue to start developing one of the main screens for data virtualization. Sept 27th is when he needs to deliver the functionality based on this component. He says "we can work with a developer to fine-tune to our requirements. we will enter the issues we are facing on the JIRA."

**Comment by seanforyou23** [comment]
This work item has been moved into our roadmap starting with finalizing the designs and implementing the html/css solution first. Closing this issue in favor of [the effort in core](https://github.com/patternfly/patternfly-next/issues/404). Will create a new issue for the react implementation once we're ready for it.
